src/main/java/com/microsoft/azure/spring/data/documentdb/DocumentDbFactory.java

```diff
@@ -11,38 +11,72 @@
 import com.microsoft.azure.documentdb.DocumentClient;
 import com.microsoft.azure.spring.data.documentdb.common.GetHashMac;
 import com.microsoft.azure.spring.data.documentdb.common.PropertyLoader;
+import com.microsoft.azure.spring.data.documentdb.common.TelemetryProperties;
+import com.microsoft.azure.spring.data.documentdb.common.TelemetryProxy;
 import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+
+import java.util.HashMap;
+import java.util.Map;
 
 public class DocumentDbFactory {
 
+    private DocumentClient documentClient;
+    private final TelemetryProxy telemetryProxy;
     private static final String USER_AGENT_SUFFIX = Constants.USER_AGENT_SUFFIX + PropertyLoader.getProjectVersion();
 
-    private DocumentClient documentClient;
+    private String getUserAgentSuffix(boolean isBiEnabled) {
+        String suffix = ";" + USER_AGENT_SUFFIX;
 
-    public DocumentDbFactory(String host, String key) {
-        this(host, key, true);
+        if (isBiEnabled && GetHashMac.getHashMac() != null) {
+            suffix += ";" + GetHashMac.getHashMac();
+        }
+
+        return suffix;
     }
 
-    public DocumentDbFactory(String host, String key, boolean isBiEnabled) {
+    public DocumentDbFactory(String host, String key) {
         Assert.hasText(host, "host must not be empty!");
         Assert.hasText(key, "key must not be empty!");
 
+        final boolean isBiEnabled = this.isTelemetryAllowed();
         final ConnectionPolicy policy = ConnectionPolicy.GetDefault();
 
-        String userAgent = ";" + USER_AGENT_SUFFIX;
-        if (isBiEnabled && GetHashMac.getHashMac() != null) {
-            userAgent += ";" + GetHashMac.getHashMac();
-        }
-        policy.setUserAgentSuffix(userAgent);
+        policy.setUserAgentSuffix(getUserAgentSuffix(isBiEnabled));
+
+        this.documentClient = new DocumentClient(host, key, policy, ConsistencyLevel.Session);
+        this.telemetryProxy = new TelemetryProxy(isBiEnabled);
 
-        documentClient = new DocumentClient(host, key, policy, ConsistencyLevel.Session);
+        this.trackCustomEvent();
     }
 
     public DocumentDbFactory(DocumentClient client) {
+        final boolean isBiEnabled = this.isTelemetryAllowed();
+
+        if (client != null && client.getConnectionPolicy() != null) {
+            client.getConnectionPolicy().setUserAgentSuffix(this.getUserAgentSuffix(isBiEnabled));
+        }
+
         this.documentClient = client;
+        this.telemetryProxy = new TelemetryProxy(this.isTelemetryAllowed());
+        this.trackCustomEvent();
+    }
+
+    private boolean isTelemetryAllowed() {
+        // panli: will allow user to disable this from application.properties in another PR.
```

Comment:
configurable telemetry collecting is must for GDPR. i'll fwd you email from bi team about it. pls implement this switch.